sk-driver/src/runner.rs

```diff
@@ -162,7 +162,16 @@ pub async fn run_trace(ctx: DriverContext, client: kube::Client) -> EmptyResult
         }
 
         if let Some(next_ts) = maybe_next_ts {
-            let sleep_duration = max(0, next_ts - sim_ts);
+            let simulation_normal_step_duration = next_ts - sim_ts;
+            let simulation_normal_step_duration_u64 = simulation_normal_step_duration.try_into().unwrap_or(u64::MAX);
+
+            let sleep_duration = if speed <= 0.0 {
+                simulation_normal_step_duration_u64
+            } else {
+                (simulation_normal_step_duration_u64 as f64 / speed).round() as u64
+            };
+
+            let sleep_duration = max(0, sleep_duration);
```

Comment:
I think we can simplify this whole thing significantly:

```
let simulation_normal_step_duration = max(0, next_ts - sim_ts) as f64;
let sleep_duration = (simulation_normal_step_duration / speed).round() as u64;
```

That way we ensure that even if there's a negative step duration, that event will happen "right now"; and speed should never be negative at this point (ideally we'd have some validation here, but that can be done in a different PR).